source/geod24/concurrency.d

```diff
@@ -849,6 +854,51 @@ struct ThreadInfo
     }
 }
 
+
+/***************************************************************************
+
+    Thread with ThreadInfo,
+    This is implemented to avoid using global variables.
+
+***************************************************************************/
+
+public class InfoThread : Thread
+{
+    public ThreadInfo info;
```

Comment:
It is never set though ?

Reply:
It's accessed through thisInfo.
Doesn't it have to be initialized?

Reply:
It does have to be initialized.

Reply:
Its initialization was implemented by ThreadScheduler.spawn.

Reply:
`ThreadScheduler.spawn` is never used. In fact, I'm considering removing `ThreadScheduler` completely, there's nothing special about it. `_spawn` is the one that gets used.

Reply:
I see. _spawn already has initialization.

Reply:
Yes it should just be replaced so that the global can be removed. Otherwise this is just dead code.

Reply:
I returned the ThreadScheduler's code to its original form.

Reply:
````D
static @property ref thisInfo() nothrow
    {
        static ThreadInfo val;
        return val;
    }
````
Global variables are used here.
When using InfoThread, the global variables were not used.
However, the main threads and other threads use global variables.